sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -407,6 +423,12 @@ object StateStore extends Logging {
     loadedProviders.remove(storeProviderId).foreach(_.close())
   }
 
+  /** Unload all state store providers: unit test purpose */
+  private[sql] def unloadAll(): Unit = loadedProviders.synchronized {
```

Comment:
Ditto, if we eagerly check the `stateSchemaCheckEnabled` config, then this test specific function also can be removed and we can use the config to control the behavior in the test?

Reply:
same here as well.